cpp/src/arrow/testing/gtest_util.h

```diff
@@ -28,6 +28,7 @@
 #include <utility>
 #include <vector>
 
+#include <gmock/gmock-matchers.h>
```

Comment:
Usual concern: does it blow up compile times?
(`gtest.h` is already a large inclusion that gets processed for every test file)

Reply:
I'll move these to `arrow/testing/matchers.h`, then